codegenerator/cli/src/config_parsing/entity_parsing.rs

```diff
@@ -603,9 +635,122 @@ impl Field {
+                     type. Field '{}'",
+                    field.name
+                ));
+            }
+            if decimal_precision_directive.arguments.len() != 1 {
+                return Err(anyhow!(
+                    "EE216: The precision directive on a BigInt should only take a single integer \
+                     argument called 'digits'. Field '{}'",
+                    field.name
+                ));
+            }
+            let (arg_name, arg_value) = decimal_precision_directive.arguments.first().unwrap();
+            if arg_name != "digits" {
+                return Err(anyhow!(
+                    "EE216: The precision directive on a BigInt should only have a 'digits' \
+                     parameter. Unknown parameter '{}'. Field '{}'",
+                    arg_name,
+                    field.name
+                ));
+            }
+            let precision = get_positive_integer(arg_value, &field.name, "digits")?;
+            pg_type_modifications.big_int_precision = Some(precision);
+        }
+
+        // Process @numeric
+        if let Some(numeric_directive) = numeric_directives.first() {
+            if !matches!(underlying_scalar, GqlScalar::BigDecimal(_)) {
+                return Err(anyhow!(
+                    "EE215: The numeric directive on a field is only suitable for BigDecimal \
+                     scalar type. Field '{}'",
+                    field.name
+                ));
+            }
+            let mut precision: Option<u32> = None;
+            let mut scale: Option<u32> = None;
+
+            for (arg_name, arg_value) in &numeric_directive.arguments {
+                match arg_name.as_str() {
+                    "precision" => {
+                        precision =
+                            Some(get_positive_integer(arg_value, &field.name, "Precision")?);
+                    }
+                    "scale" => {
+                        scale = Some(get_positive_integer(arg_value, &field.name, "Scale")?);
+                    }
+                    unknown_param => {
+                        return Err(anyhow!(
+                            "EE216: The numeric directive on a BigDecimal should only have \
+                             'precision' and 'scale' parameters. Unknown parameter '{}'. Field \
+                             '{}'",
+                            unknown_param,
+                            field.name
+                        ));
+                    }
+                }
+            }
+            if precision.is_none() || scale.is_none() {
+                return Err(anyhow!(
+                    "EE216: The numeric directive on a BigDecimal must have both 'precision' and \
+                     'scale' parameters. Field '{}'",
+                    field.name
+                ));
+            }
+            pg_type_modifications.big_decimal_precision_scale =
+                Some((precision.unwrap(), scale.unwrap()));
```

Comment:
Logically I've looked through your validation and can see that the unwraps are safe here. But in general I prefer if we lean on pattern matching and avoid unwraps as it makes refactors a lot more safe.

Reply:
Done :)